lib/smart_name.rb

```diff
@@ -26,7 +33,12 @@ class SmartName < Object
   class << self
     def new obj
       return obj if obj.is_a? self.class
-      str = obj.is_a?(Array) ? obj * joint : obj.to_s
+      str =
+        if obj.is_a?(Array)
+          obj.map(&:to_s) * joint
+        else
+          obj.to_s
+        end
```

Comment:
this (very important) method is getting a little ugly.  maybe we need a `stringify` method or something?